### PR TITLE
[Fix #1061] Add cider-find-ns

### LINKED ISSUE #1061: Add cider-find-ns
Here are some very common situations for me:

1. I know some function exists, but I don't recall the exact name.
2. I think maybe some function exists in `lib` and I want to check before I write it myself

I'm solving these now by using `cider-classpath`, `C-s` down to the `lib`, open the jar, look at the list of files and then picking something that looks promising.

`cider-find-ns` would give a completing read of all the available namespaces available on classpath, not just in my own project, and jumps to the selected ns.

### PR DESCRIPTION
**Pull request by expez**
Provide a completing read of available namespaces and jump to the file
containing its definition.

### CODE REVIEW

File: cider-interaction.el [hunk 1]
```diff
@@ -924,6 +924,31 @@ thing at point."
                  #'cider--find-var-other-window
                #'cider--find-var))))
 
+(defun cider-sync-request:ns-path (ns)
+  "Toggle namespace tracing for NS."
```
Comment (bbatsov) [inline cider-interaction.el]: Really? :-)

File: cider-interaction.el [hunk 2]
```diff
@@ -924,6 +924,31 @@ thing at point."
...
+  "Find the file containing NS.
+
+A prefix of `-` or a double prefix argument causes
+the results to be displayed in a different window."
+  (interactive "P")
+  (cider-ensure-op-supported "ns-path")
+  (if ns
+      (cider--find-ns ns)
+    (let* ((namespaces (cider-sync-request:ns-list))
+           (ns (completing-read "Namespace: " namespaces)))
```
Comment (bbatsov) [inline cider-interaction.el]: I think the prompt should say `Find namespace: `.

File: cider-interaction.el [hunk 3]
```diff
@@ -924,6 +924,31 @@ thing at point."
...
+            "ns" ns)
+      nrepl-send-sync-request
+      (nrepl-dict-get "path")))
+
+(defun cider--find-ns (ns &optional other-window)
+  (-if-let (path (cider-sync-request:ns-path ns))
+      (cider-jump-to (cider-find-file path) nil other-window)
+    (error "Can't find %s" ns)))
+
+(defun cider-find-ns (&optional arg ns)
```
Comment (bbatsov) [inline cider-interaction.el]: This command could benefit from some keybinding.

Comment (expez) [inline cider-interaction.el]: Yes, but I couldn't think of anything obvious.  Any ideas?

**Comment by expez** [comment]
I've addressed your concerns

File: cider-interaction.el (hunk 3 above)
Comment (bbatsov) [inline cider-interaction.el]: Something free. :-) Ideally it'd resemble `M-.` as it serves a similar purpose. Have to think a bit more about this.

Comment (expez) [inline cider-interaction.el]: `C-c C-.` is unbound for me.

Comment (bbatsov) [inline cider-interaction.el]: Sounds good. Bind it in both modes and mention it in the README.

Comment (expez) [inline cider-interaction.el]: Done, also updated the changelog with the new keybinding.

**Pull request merged**